### ISSUE TITLE
Build fails when using TypeScript <3.7 - Breaking change in transitive dependency?

### ISSUE THREAD

**Issue by TaylorPzreal** [formatjs/react-intl#1548]
I use react-intl@3.4.0, but when i npm i, it upgrade dependency packages version at package-lock.json.
There has an error after i execute npm run build:

```
node_modules/@formatjs/intl-utils/dist/invariant.d.ts:1:92 - error TS1144: '{' or ';' expected.

1 export declare function invariant(condition: boolean, message: string, Err?: any): asserts condition;

```

### Expected behavior

react-intl should use library with fixed version to avoid some error.

### Current behavior
build error

### My Environment
react@16.8.6
react-intl@3.4.0
npm@6.11.3
node@12.10.0
typescript@3.2.4

macOS Catalina10.15.2 Beta (19C46a)

**Comment by vyacheslav-pushkin** [comment]
We were hit by this too.

Basically the problem is a breaking change in newer patch versions of several FormatJS libraries (such as `@formatjs/intl-relativetimeformat`, `@formatjs/intl-listformat`, etc.) which now depend on a newer major version of `@formatjs/intl-utils`. Which, in turn, requires TypeScript 3.7+, as it uses `asserts condition`.

**package-lock before:**

```
    "@formatjs/intl-relativetimeformat": {
      "version": "4.5.1",
      "resolved": "https://registry.npmjs.org/@formatjs/intl-relativetimeformat/-/intl-relativetimeformat-4.5.1.tgz",
      "integrity": "sha512-s32xwuxvkozX6NxUVAnEGZU1/k3HA3kSsRXtpCvftnCWsshNAUsUWLS/6GpfHcjLfY1XEGptSKouTasyGLZdsg==",
      "requires": {
        "@formatjs/intl-utils": "^1.6.0"
      }
    },
    . . .
```

**package-lock now:**

```
    "@formatjs/intl-relativetimeformat": {
      "version": "4.5.3",
      "resolved": "https://registry.npmjs.org/@formatjs/intl-relativetimeformat/-/intl-relativetimeformat-4.5.3.tgz",
      "integrity": "sha512-j8ZGaS0Rcx0nDG+rnsDUnw7p0DH1pBV+pz3LIm4eZRl1f93IBC/RbR8E1mAvT28kAyq4vVg5YNDGI3oUlpbR9g==",
      "requires": {
        "@formatjs/intl-utils": "^2.0.0"
      },
      . . .
```

**Comment by vyacheslav-pushkin** [comment]
@TaylorPzreal perhaps it would be useful to change the issue title to something along the lines of `Build fails when using TypeScript <3.7 - Breaking change in transitive dependency?`. Right now I think it can be misunderstood for a (less severe) problem with TS typings.

**Comment by longlho** [comment]
Sorry about that I didn't realize `lerna` doesn't bump transitive deps properly. Published `3.9.3` which is the latest version that we'll support TS 3.7- so you guys can pin to that.
